app/models/piece.rb

```diff
@@ -44,57 +44,58 @@ def is_move_obstructed?(new_rank, new_file)
+        return true if piece_at_offset.present?
       end
       false
     elsif new_rank < rank && new_file > file
       (new_rank - rank..-1).each do |offset|
         piece_at_offset = current_pieces.select{|piece| piece.rank == rank + offset && piece.file == file - offset}
         next if exception_for_capture.call(offset, new_rank - rank, piece_at_offset)
-        return true if piece_at_offset.present?        
+        return true if piece_at_offset.present?
       end
       false
 
-    # Vertical moves     
-    elsif new_rank > rank 
+    # Vertical moves
+    elsif new_rank > rank
       (1..new_rank - rank).each do |offset|
         piece_at_offset = current_pieces.select{|piece| piece.rank == rank + offset && piece.file == file}
         next if exception_for_capture.call(offset, new_rank - rank, piece_at_offset)
         return true if piece_at_offset.present?
       end
       false
-    elsif new_rank < rank 
+    elsif new_rank < rank
       (new_rank - rank..-1).each do |offset|
         piece_at_offset = current_pieces.select{|piece| piece.rank == rank + offset && piece.file == file}
         next if exception_for_capture.call(offset, new_rank - rank, piece_at_offset)
-        return true if piece_at_offset.present?        
+        return true if piece_at_offset.present?
       end
       false
 
     # Horizontal moves
     elsif new_file > file
       (1..new_file - file).each do |offset|
         piece_at_offset = current_pieces.select{|piece| piece.rank == rank && piece.file == file + offset}
-        next if exception_for_capture.call(offset, new_file - file, piece_at_offset) 
-        return true if piece_at_offset.present?         
+        next if exception_for_capture.call(offset, new_file - file, piece_at_offset)
+        return true if piece_at_offset.present?
       end
       false
     elsif new_file < file
       (new_file - file..-1).each do |offset|
         piece_at_offset = current_pieces.select{|piece| piece.rank == rank && piece.file == file + offset}
-        next if exception_for_capture.call(offset, new_file - file, piece_at_offset) 
-        return true if piece_at_offset.present?  
+        next if exception_for_capture.call(offset, new_file - file, piece_at_offset)
+        return true if piece_at_offset.present?
       end
       false
     end
   end
-  
+
+  # This method checks if a player tries to move a piece
+  # back to the same square after pick up or tries to move
+  # a piece outside of the board.
   def is_move_malformed?(new_rank, new_file)
-      return true if rank == new_rank && file == new_file
-      return true if rank < 1 || rank > 8
-      return true if file < 1 || file > 8
-      return true if new_rank < 1 || new_rank > 8
-      return true if new_file < 1 || new_file > 8
+    return true if rank == new_rank && file == new_file
+    return true if new_rank < 1 || new_rank > 8
+    return true if new_file < 1 || new_file > 8
+    false
```

Comment:
I removed the check for rank and file and added a return of false if the move is not malformed.